services/management/backup/backups_service.go

```diff
@@ -18,116 +18,356 @@ package backup
+			PMMServerConfig: location.PMMServerConfig,
+			PMMClientConfig: location.PMMClientConfig,
+			S3Config:        location.S3Config,
+		}
+
 		var dataModel models.DataModel
 		var jobType models.JobType
-		switch svc.ServiceType {
+		switch csr.ServiceType {
 		case models.MySQLServiceType:
 			dataModel = models.PhysicalDataModel
 			jobType = models.MySQLBackupJob
 		case models.MongoDBServiceType:
 			dataModel = models.LogicalDataModel
 			jobType = models.MongoDBBackupJob
-		case models.PostgreSQLServiceType,
-			models.ProxySQLServiceType,
-			models.HAProxyServiceType,
-			models.ExternalServiceType:
-			return status.Errorf(codes.Unimplemented, "unimplemented service: %s", svc.ServiceType)
 		default:
-			return status.Errorf(codes.Unknown, "unknown service: %s", svc.ServiceType)
+			panic(fmt.Sprintf("unexpected ServiceType %v", csr.ServiceType))
 		}
 
 		artifact, err = models.CreateArtifact(tx.Querier, models.CreateArtifactParams{
 			Name:       req.Name,
-			Vendor:     string(svc.ServiceType),
+			Vendor:     string(csr.ServiceType),
+			Version:    dbVersion,
 			LocationID: location.ID,
-			ServiceID:  svc.ServiceID,
+			ServiceID:  req.ServiceId,
 			DataModel:  dataModel,
 			Status:     models.PendingBackupStatus,
 		})
 		if err != nil {
 			return err
 		}
 
-		job, config, err = s.prepareBackupJob(tx.Querier, svc, artifact.ID, jobType)
+		job, err = s.prepareBackupJob(tx.Querier, csr.AgentID, artifact.ID, jobType)
 		if err != nil {
 			return err
 		}
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+
+	switch csr.ServiceType {
+	case models.MySQLServiceType:
+		err = s.jobsService.StartMySQLBackupJob(job.ID, job.PMMAgentID, 0, req.Name, csr.DBConfig, locationConfig)
+	case models.MongoDBServiceType:
+		err = s.jobsService.StartMongoDBBackupJob(job.ID, job.PMMAgentID, 0, req.Name, csr.DBConfig, locationConfig)
+	default:
+		panic(fmt.Sprintf("unexpected ServiceType %v", csr.ServiceType))
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	return &backupv1beta1.StartBackupResponse{
+		ArtifactId: artifact.ID,
+	}, nil
+}
+
+func (s *BackupsService) findCompatibleServices(
+	serviceType models.ServiceType,
+	artifactDBVersion string,
+) ([]*models.Service, error) {
+	if artifactDBVersion == "" {
+		s.l.Info("skip finding compatible services: artifact db version is unknown")
+		return []*models.Service{}, nil
+	}
+
+	filters := models.ServiceFilters{
+		ServiceType: &serviceType,
+	}
+	filteredServices, err := models.FindServices(s.db.Querier, filters)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(filteredServices) == 0 {
+		return []*models.Service{}, nil
+	}
+
+	const maxWorkers = 5
+	workers := len(filteredServices)
+	if workers > maxWorkers {
+		workers = maxWorkers
+	}
+
+	g := &errgroup.Group{}
+	// stream service for checking
+	serviceCh := make(chan *models.Service, workers)
+	g.Go(func() error {
+		defer close(serviceCh)
+
+		for _, s := range filteredServices {
+			serviceCh <- s
+		}
+
```

Comment:
Why not just write all available messages to the channel? Why we need this channel length limitation?